src/dex/fluid-dex/fluid-dex.ts

```diff
@@ -0,0 +1,881 @@
+            colReserves: state.collateralReserves,
+            debtReserves: state.debtReserves,
+            exchange: this.dexKey,
+          },
+          exchange: this.dexKey,
+          poolIdentifier: pool.id,
+          gasCost: FLUID_DEX_GAS_COST, // to be done
+          poolAddresses: [pool.address],
+        },
+      ];
+    } catch (e) {
+      this.logger.error(
+        `Error_getPricesVolume ${srcToken.address || srcToken.symbol}, ${
+          destToken.address || destToken.symbol
+        }, ${side}:`,
+        e,
+      );
+
+      return null;
+    }
+  }
+
+  // Returns estimated gas cost of calldata for this DEX in multiSwap
+  getCalldataGasCost(poolPrices: PoolPrices<FluidDexData>): number | number[] {
+    return CALLDATA_GAS_COST.DEX_NO_PAYLOAD;
+  }
+
+  // Encode params required by the exchange adapter
+  // V5: Used for multiSwap, buy & megaSwap
+  // V6: Not used, can be left blank
+  // Hint: abiCoder.encodeParameter() could be useful
+  getAdapterParam(
+    srcToken: string,
+    destToken: string,
+    srcAmount: string,
+    destAmount: string,
+    data: FluidDexData,
+    side: SwapSide,
+  ): AdapterExchangeParam {
+    if (side === SwapSide.BUY) throw new Error(`Buy not supported`);
+    const { exchange } = data;
+
+    // Encode here the payload for adapter
+    const payload = '';
+
+    return {
+      targetExchange: exchange,
+      payload,
+      networkFee: '0',
+    };
+  }
+
+  // This is called once before getTopPoolsForToken is
+  // called for multiple tokens. This can be helpful to
+  // update common state required for calculating
+  // getTopPoolsForToken. It is optional for a DEX
+  // to implement this
+  async updatePoolState(): Promise<void> {
+    this.initializePricing(
+      await this.dexHelper.web3Provider.eth.getBlockNumber(),
+    );
+  }
+
+  // Returns list of top pools based on liquidity. Max
+  // limit number pools should be returned.
+  async getTopPoolsForToken(
+    tokenAddress: Address,
+    limit: number,
+  ): Promise<PoolLiquidity[]> {
+    const latestBlockNumber_ =
+      await this.dexHelper.web3Provider.eth.getBlockNumber();
+    let liquidityAmounts: { [id: string]: bigint } = {};
+    for (const pool of this.pools) {
+      if (
+        pool.token0 === tokenAddress.toLowerCase() ||
+        pool.token1 === tokenAddress.toLowerCase()
+      ) {
+        const state: FluidDexPoolState = await this.eventPools[
+          pool.id
+        ].getStateOrGenerate(latestBlockNumber_, false);
+
+        liquidityAmounts[pool.id] =
+          pool.token0 === tokenAddress
+            ? state.collateralReserves.token0RealReserves +
+              state.debtReserves.token0RealReserves
+            : state.collateralReserves.token1RealReserves +
+              state.debtReserves.token1RealReserves;
+      }
+    }
+
+    const entries = Object.entries(liquidityAmounts);
+
+    // Sort the entries based on the values in descending order
+    entries.sort((a, b) => {
```

Comment:
does order matter here for top pools? then we should sort by USD value not by token value here? 

1 ETH would be worth much more than 1 DAI, yet 2 DAI would be sorted higher with this than 1 ETH

Reply:
@bergben this function is intended to return k number of pools with most liquidity of a specific token.
This sort is to create an array whose first k entries are relevant(i.e. have the max liquidity).

Reply:
imo we need liquidity for a specific token in the pool and not the liquidity of all the tokens in the pool, so in line 348, I have assigned the liquidity of the token in the pool to the `liquidityAmounts` array.